### ISSUE TITLE
Add support for singularizing entities

### ISSUE THREAD

**Issue by daniel-gato** [RobinBlomberg/kysely-codegen#32]
Hi,

For context: It is quite common to have tables name in their plural form to describe what the table contains. This is the approach that Ruby on Rails takes as well. Our model Car, will be stored in a "cars" table.

Would it be possible to have a "singularise-entities" to turn:

```typescript
export interface Users {
  id: string;
  email: string;
}

export interface Database {
  users: Users;
}
```

into

```typescript
export interface User {
  id: string;
  email: string;
}

export interface Database {
  users: User;
}
```

The same way that camelCase works? 

Another reason to do that is to avoid having mixed types with reserved table names. `User` table name is not allowed in PG for example. In the same example, from memory, mysql won't allow table names like `admin`, `backup`, `channel`, `action`

## Upvote & Fund

- We're using [Polar.sh](https://polar.sh/RobinBlomberg) so you can upvote and help fund this issue.
- We receive the funding once the issue is completed & confirmed by you.
- Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/RobinBlomberg/kysely-codegen/issues/32">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/RobinBlomberg/kysely-codegen/issues/32/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/RobinBlomberg/kysely-codegen/issues/32/pledge.svg">
</picture>
</a>

**Comment by RobinBlomberg** [comment]
This is an interesting idea. I would probably use [pluralize](https://www.npmjs.com/package/pluralize) for this.

I think, if this is implemented, we will sooner or later need a way to provide custom singularization rules as well. In that case, we might also need configuration capabilities as in #30.

**Comment by acro5piano** [comment]
Created a PR for this: https://github.com/RobinBlomberg/kysely-codegen/pull/162

**Comment by RobinBlomberg** [comment]
Should be released in [kysely-codegen@0.16.0](https://github.com/RobinBlomberg/kysely-codegen/releases/tag/0.16.0)! 🚀
